grammars/tree-sitter-ruby.cson

```diff
@@ -154,6 +158,7 @@ scopes:
   'call > identifier:nth-child(2)': 'entity.name.function'
   'method_call > identifier:nth-child(0)': [
     {exact: 'require', scopes: 'support.function'}
+    {match: '^(public|protected|private)$', scopes: 'keyword.other.special-method'}
```

Comment:
:wave: @maxbrunsfeld: If you have a moment, can you take a look at this Tree-sitter-related question?

These visibility modifiers can appear in two different scenarios (as seen in the [pull request body](https://github.com/atom/language-ruby/pull/269#issue-304350924)). For example:

```rb
private # standalone visibility modifier

private def foo; end # method-specific visibility modifier
```

We want syntax highlighting to behave the same in both scenarios. We initially thought we could handle both scenarios just by adding lines 140-143 here:

https://github.com/atom/language-ruby/blob/8e38e74e75246c659e7b4a9adf88a510d980b3e9/grammars/tree-sitter-ruby.cson#L135-L144

However, later in the grammar file, there's a more specific selector that matches the method-specific visibility modifier:

https://github.com/atom/language-ruby/blob/952e75bf9481110adf3891773500d11c0c3ddb38/grammars/tree-sitter-ruby.cson#L155-L158

So, if we _only_ have the change seen in lines 140-143, then `private def foo; end` tokenizes `private` as `entity.name.function`. 

To address that issue, we essentially duplicated the logic in 140-143 and added it here:

https://github.com/atom/language-ruby/blob/8e38e74e75246c659e7b4a9adf88a510d980b3e9/grammars/tree-sitter-ruby.cson#L159-L163

Does this seem like a reasonable approach? Or is there another way that feels cleaner?

Reply:
Yeah, I think that is reasonable. The selector matching that we implemented isn’t super powerful and expressive, so I think some duplication is ok.

Reply:
Thanks for the consultation, @maxbrunsfeld. :zap::bow: